lib/dry/view/decorator.rb

```diff
@@ -0,0 +1,32 @@
+require 'dry/core/inflector'
+require 'dry/view/part'
+
+module Dry
+  module View
+    class Decorator
+      attr_reader :config
+
+      # @api public
+      def call(name, value, renderer:, context:, **options)
+        klass = part_class(name, value, **options)
+
+        if value.respond_to?(:to_ary)
+          singular_name = Dry::Core::Inflector.singularize(name).to_sym
+
+          arr = value.to_ary.map { |obj|
+            call(singular_name, obj, renderer: renderer, context: context, **options)
+          }
+
+          klass.new(name: name, value: arr, renderer: renderer, context: context)
+        else
+          klass.new(name: name, value: value, renderer: renderer, context: context)
+        end
+      end
+
+      # @api public
+      def part_class(name, value, **options)
+        options.fetch(:as) { Part }
+      end
```

Comment:
Ah, so the idea here was to expose a public method (`#part_class`) that application-specific subclasses of the decorator could use to provide resolution of custom part classes, without having to go to the length of overriding the basic behaviour in `#call`. This is the sort of thing I'd want to put in API docs for this method.

So in this case (with the default `#call` implementation left as is), there will _always_ be a `name` and `value` passed to this method, and `**options` is anything that's provided as the options to the corresponding `expose` in the view controller. Because these are distinct bits of data coming from 2 different places, I don't want to smoosh them together into a single options hash, especially since it could potentially result in options being lost (or errors unexpectedly being raised) in the case of naming collisions.

Does that make sense?